### PR TITLE
Archisinal_NFT_Marketplace.md

### PR DESCRIPTION
**Pull request by Archisinal**

### CODE REVIEW

**Comment by TtomaS7** [comment]
Hello @Archisinal 
Thank you for your interest and for submitting your application. We truly appreciate your effort!

I'll assign the curator for the review.

File: Archisinal.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+### **Milestone 1: Smart Contract Development**
+
+- **Estimated duration:** 1 month
+- **FTE:** 3
+- **Costs:** 24,000 USD
+
+**Overview:**
+
+- Develop smart contracts for NFT minting, buying, selling, and royalty distribution
```
Comment (h4x3rotab) [inline Archisinal.md:192]: Here you propose to create your own NFT platform. However, building infrastructure requires a lot of time, and the maintenance is even more costly. Given the mission of the team is to create a marketplace dedicated for AEC, it looks to me a better idea is to build your project on top of some existing NFT projects (for example, [RMRK ink!](https://github.com/rmrk-team/rmrk-ink)). In this way, you can focus on your own market and domain knowledge rather than building an infrastructure-heavy project.

The benefit of using a shared infra is huge:

1. You save a good amount of time to build your own unique features and UX
2. You can utilize all the future innovation & updates from the infra project
3. You can get additional traffic from the infra community
4. You will build a strong relationship in the general ink! ecosystem instead of becoming a competitor of other infra team

Have you considered this option?

Comment (Archisinal) [reply Archisinal.md:192]: Hi!

Thanks for taking the time to review the application. I will let Danil weigh in with his past experiences regarding using shared infrastructure, but I can add my own thoughts from a general perspective.

The NFT marketplace is the MVP of a project which will eventually morph into a dispute and arbitration platform for professionals in the construction and architecture industry. The technical requirements for this platform will be continuously morphing, integrating DAO functionality, DIDs and payment automation- this will require connection to Polkadot's various parachains. For these reasons, I feel it important that Archisinal is built from scratch with it's own code base and not reliant on other protocol standards.

Furthermore, I will be speaking at the [Construction Blockchain Consortium](https://www.constructionblockchain.org/) in Berlin in August, and would like to use it as an opportunity to highlight the benefits of building and hosting a Dapp on Astar from scratch. The community/target audience for this project will be commercial industry professionals as opposed to web3 native NFT, so I would like to build a sovereign platform which is more tailored to that demographic.

Comment (sventime) [reply Archisinal.md:192]: Hi @h4x3rotab! 

I appreciate your valid remarks. There is no great variety of NFT infrastructure for ink! nowadays, but RMRK is definitely the most prominent and suitable. Let me add my thoughts from a tech perspective on why it may be not the best choice for our project:

1. The RMRK standard far exceeds our requirements. Its main features such as nesting, equippability, and multi-resources, are not relevant to our use cases. Our needs align more closely with the basic NFT features provided by PSP34, which RMRK ink! actually reuses in their implementation.
2. Currently, there doesn't seem to be an RMRK ink! marketplace implementation. There is for sure rmrk-pallet for substrate, but not for ink!

Taking the above into account the integration with RMRK at this point, especially for MVP can make more difficulties than benefits. I can tell this from my previous experience with [Asylum](https://github.com/w3f/Grants-Program/blob/master/applications/asylum.md).
For sure, we won't build everything from scratch and we intend to leverage tools such as OpenBrush, Swanky, and Typechain.

File: Archisinal.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
+## Archisinal - NFT Marketplace ##
+
+- **Team Name:** Archisinal Technologies
+- **Payment Address:** 14xeAxD7LSJQmNXn2RQcm6WTJiByv6HKn6T8fTjqX1HeCMwR
+
+## Project Overview :page_facing_up:
+### Overview
+
+- Archisinal wants to become the Opensea for professionals in Architecture, Engineering and Construction (AEC).
```
Comment (DoubleOTheven) [inline Archisinal.md:9]: This is a very unique and intriguing idea. As a former architect, I support this :)

Comment (Archisinal) [reply Archisinal.md:9]: This is great news, Sam! We should have a call - always looking for feedback from different industry perspectives.

**Comment by SotaWatanabe** [comment]
Thank you very much for your comprehensive proposal. I want to support this proposal. One thing I want to double check is that all smart contracts and deliberations will be delivered with 100% open source licence. Is that ok for your team?

**Comment by Archisinal** [comment]
> Thank you very much for your comprehensive proposal. I want to support this proposal. One thing I want to double check is that all smart contracts and deliberations will be delivered with 100% open source licence. Is that ok for your team?

Hi @SotaWatanabe - appreciate the support. Yes confirming all sc and deliberatiosn will be 100% open source.

**Review by SotaWatanabe — APPROVED** [review]

File: Archisinal.md (hunk 1 above)
Comment (DoubleOTheven) [reply Archisinal.md:192]: Good feedback. We discussed internally and agree that the ecosystem is young. This is a good time for you guys to explore and address your unique business goals.

File: Archisinal.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+- Through on-chain tokenization of designs, professionals in the AEC industry can reduce the quantity and length of arbitration disputes and gain exposure to the creator economy through royalties from secondary sales.
+- The purpose of this document is to provide an executive summary for a template aimed at building an innovative NFT (Non-Fungible Token) marketplace for architects and engineers. This platform will revolutionize the way professionals in the architectural and engineering sectors trade, showcase, and monetize their digital assets, whilst enabling a global audience to access and acquire unique pieces of digital art and design.
+- Whilst both Ink! and Solidity are popular programming languages for building smart contracts and decentralized applications, there are some key reasons why Ink! may be better suited for building the Archisinal NFT marketplace:
```
Comment (DoubleOTheven) [inline Archisinal.md:12]: Nit: `Ink!` -> `ink!`

**Review by DoubleOTheven — APPROVED** [review]
Solid full stack team, and interesting idea. I'm excited to see this one.

**Comment by 0xMarkian** [comment]
Hey, thanks for the application! I love the idea. One clarification - is this an application under the business category? if that is the case could you also share your pitch deck/business model ?

File: Archisinal.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+
+## Additional Information :heavy_plus_sign:
+
+**How did you hear about the Bounty Program?** Announcement by another team
+
+- The Astar Grants program was highlighted through a Medium blogpost.
+- Basic POC entry in the Encode Club Spring 2023 Hackathon - [Here](https://github.com/Archisinal/Encode_Hackathon)
```
Comment (HCastano) [inline Archisinal.md:307]: This link 404s

File: Archisinal.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+- Testing and deployment on Astar testnet Shibuya
+
+| Number | Deliverable | Specification |
+| --- | --- | --- |
+| 0a. | License | MIT |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how users can interact with the developed smart contracts. |
+| 0c. | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit & integration tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an article that explains the development and features of the smart contracts, as well as any challenges faced and lessons learned. |
+| 1. | NFT Minting | Develop a smart contract functionality for NFT minting when creators upload their designs. |
```
Comment (HCastano) [inline Archisinal.md:202]: Where are you going to store all the metadata associated with the NFTs?

File: Archisinal.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+| 2. | NFT Transfer | Develop a smart contract functionality for NFT transfer when a sale happens, including royalty distribution. |
+| 3. | Creator’s Administering | Develop a smart contract functionality for creators, including managing their project NFTs data and tracking sales. |
+| 4. | Buyer’s Administering | Develop a smart contract functionality for buyers, including tracking their purchases and NFT holdings. |
+| 5. | Marketplace Administration | Develop a smart contract functionality for marketplace administrators, including the ability to delist inappropriate items. |
+| 6. | Auction | Develop smart contract functionality for auctioning NFTs, allowing users to place bids and manage their auctions. |
+| 7. | Test Network Deployment | Deploy the developed smart contracts on a test network (Shibuya) for evaluation and testing. |
+| 8. | Quality, Performance, and Security Considerations | Ensure that the developed smart contracts meet quality, performance (weight), and security requirements. |
```
Comment (HCastano) [inline Archisinal.md:209]: How do you plan to test these things?

File: Archisinal.md [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+    - MVP: 2.5 months
+    - Post-MVP: 9 months
+    - Post-Launch: 10 months
+- **Full-Time Equivalent (FTE): 6**
+- **Total Costs:**
+    - MVP: $60,000
+
+## MVP (brief discussion)
+
+### **Milestone 1: Smart Contract Development**
```
Comment (HCastano) [inline Archisinal.md:184]: I'm not opposed to having multiple implementation of the same thing, especially in an emerging ecosystem - but why are you choosing to write your own contracts instead of building on top of something like RMRK or ArtZero?

File: Archisinal.md [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+
+**Artem Smolyarenko** - UI/UX designer with over 3 years experience. He specializes in brand development in all aspects and have an excellent understanding of typography, color psychology, as well as grid, web and mobile principles. Artem worked on different types of products with startups and corporate teams from around the world. Skills: UI/UX design, product design, web and mobile app design. Tech stack & tools: Figma, Adobe Photoshop, Adobe Illustrator, Adobe XD.
+
+### Team Code Repos
+
+- **Archisinal** [https://github.com/Archisinal](https://github.com/Archisinal)
+- **Danylo Semirazov** - [github.com/sventime](https://github.com/sventime), [gitlab.com/asylum-space](https://gitlab.com/asylum-space)
+- **Oleh Kulykovskyi**  - [github.com/blindbeat](http://github.com/blindbeat), [gitlab.com/asylum-space](https://gitlab.com/asylum-space)
```
Comment (HCastano) [inline Archisinal.md:151]: There don't seem to be a lot of contributions on GitHub for either team member. Would you be able to point to some, specially related to Rust/Substrate/Solidity?

**Review by HCastano — COMMENTED** [review]
I've got a few questions, but I'd say I'm a _soft yes_ on this. Having another NFT market implementation in ink! could be interesting, assuming that it doesn't just copy-paste from existing implementations.

The full-stack nature is also nice, since we need more code with that in mind.

File: Archisinal.md [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+
+- The objective of the Archisinal project is to build a NFT marketplace that is similar to Opensea, but tailored to the specific needs of architects, and engineers by providing them a wider audience to connect with for on- and off-chain work engagements. Archisinal wants to become the ‘go-to platform’ for architects and designers to generate more revenue by easily creating and selling unique, one-of-a-kind digital assets as NFTs on the Astar blockchain. By leveraging smart contract functionality, architects will be provided with a platform for architects to showcase and sell their designs as unique digital assets, and earn royalties from secondary sales. With its search functionality and easy buying and selling capabilities, it will also make it easy for architecture enthusiasts to discover and purchase designs that interest them.
+
+Holding the digital asset provides value in two main funnels:
+
+1) To buy, hold and sell like typical NFT art 
+
+2) Utility token - holding the token for a particular designer gives the holder token-gated access to the original creator to develop the design into a physical product
+
+**Philosophical Questioning:**
```
[the remark this answers is not in the document]
Comment (Archisinal) [reply Archisinal.md:32]: Think of it this way - the creator of the Nike logo was paid $35 in 1971. The point of the MJ house example was to highlight that some things do not hold the gravitas that they may do in the future. Whilst I imagine that royalties from secondary sales probably wouldn't be enough for this designer to retire on, it gives a supplementary source of income from work they have created which holds weight culturally.

File: Archisinal.md (hunk 8 above)
Comment (Archisinal) [reply Archisinal.md]: PR updated with contributions of team members. cc: @sventime

File: Archisinal.md (hunk 5 above)
Comment (sventime) [reply Archisinal.md:202]: We plan to implement the PSP24Metadata trait, which offers a simple key-value struct for defining metadata. Considering the specifics of the project, this metadata is not expected to change frequently. Therefore, we will primarily store the IPFS link to the project's metadata on-chain. The metadata will include the project's name, description, media file links, project source links, etc. The creator will be able to update this IPFS link after confirming with the new project owner.

File: Archisinal.md (hunk 6 above)
Comment (sventime) [reply Archisinal.md:209]: To ensure both quality and security requirements are met, we will utilize static analysis tools such as **Clippy** and **Semgrep**. The reports generated by these tools will be attached to pull requests (PRs). Further, we plan to integrate these tools with GitHub CI.

For performance requirements, we plan to implement a series of integration performance tests. These tests will primarily focus on gas limit assertions and will be executed against the Astar testnet to account for the weights of host functions. The aim of such tests is to identify and optimize problematic functions.

To address security requirements, we will develop a set of integration tests that specifically target security vulnerabilities and edge cases. These tests will cover fundamental vulnerabilities such as reentrancy and function visibility. Additionally, we will conduct a manual self-audit of our code, leveraging our expertise in ink! and Rust and adhering to best practices and smart contract requirements. While currently there are only Solidity audit tools available publicly, some of the rules can be also applicable to WASM smart contracts.

File: Archisinal.md (hunk 7 above)
Comment (sventime) [reply Archisinal.md]: Thank you for your valuable comment. I have provided my arguments about RMRK in the [comment above](https://github.com/use-inkubator/Ecosystem-Grants/pull/19#discussion_r1196634848).

And would also like to admit that [ArtZero Contracts](https://github.com/ArtZero-io/Contracts) aligns more closely with our use case, as it includes marketplace functionality, auction features, and the ability to post and edit projects. However, some of these features exceed our requirements. In Archisinal, particularly during the MVP stage, we intend to maintain a very simple flow where each NFT project is personalized for a specific client and managed by the creator of the NFT.

Considering the points mentioned above, along with the comment from @Archisinal regarding the future plans of the platform, integrating with RMRK or ArtZero at this stage, especially for the MVP, could introduce more difficulties than benefits.

File: Archisinal.md (hunk 4 above)
Comment (HCastano) [reply Archisinal.md]: Thanks for changing the link. Can you also post a video to Loom or something showing the functionality? It makes it easier (as opposed to me having to connect my wallet, get testnet tokens, etc.)

File: Archisinal.md (hunk 5 above)
Comment (HCastano) [reply Archisinal.md]: Do you mean [PSP34Metadata](https://github.com/w3f/PSPs/blob/master/PSPs/psp-34.md#psp34metadata)?

File: Archisinal.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (Archisinal) [reply Archisinal.md]: It is more about the value of the architect rising as their profile grows. Another example would be to consider Bjark Ingels, and how much the designs he created whilst studying to be an architect would sell for today.

File: Archisinal.md (hunk 4 above)
Comment (Archisinal) [reply Archisinal.md]: [Screencast from 05-08-2023 03_48_40 PM (3).webm](https://github.com/use-inkubator/Ecosystem-Grants/assets/123243302/426364eb-8abf-4aeb-ab8b-f377aed87d6a)

File: Archisinal.md (hunk 5 above)
Comment (sventime) [reply Archisinal.md]: Sure, typo). It’s PSP34Metadata.

File: Archisinal.md (hunk 6 above)
Comment (sventime) [reply Archisinal.md]: Of course, here are our plans:
a) we're going to implement a set of Typescript integration tests and leverage [typechain-polkadot](https://github.com/Brushfam/typechain-polkadot) to enhance their readability and user-friendliness.
b) polkadot.js API provides [gasConsumed](https://github.com/polkadot-js/api/blob/edde226740341396bc59bf96ad38ea559357bb9d/packages/types/src/interfaces/contracts/types.ts#L74) field, which we intend to utilize for performance tracking.

**Comment by 0xMarkian** [comment]
Hey @Archisinal ! Apologies for the delay. Returning to the discussion, I love that you are trying to tackle a real-world problem. While I am pretty confident in the tech delivery, I still have questions regarding your go-to-market strategy and business model. Please reach out on Telegram for more convenience - @OxMarkian :)

File: Archisinal.md (hunk 7 above)
Comment (Archisinal) [reply Archisinal.md]: Circling back to this @HCastano - I believe the RMRK integration is no longer feasible as they focus more on EVM solutions. From my point of view, this reinforces why we need to build out our own solution, as to not have a reliance on another team to support our tech stack.

File: applications/Archisinal.md [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+### Milestone 2: Basic UI/UX and production deployment
+
+- **Estimated duration:** 1.5 months
+- **FTE:** 3
+- **Costs:** 36,000 USD
+
+**Overview:**
+
+- Design and develop a user-friendly interface for the NFT marketplace. Technical stack: React + Next.js
+- Integrate wallet connectivity and support for multiple tokens (ETH, DOT, and USDT)
```
Comment (HCastano) [inline applications/Archisinal.md:220]: From earlier you mentioned that you wanted to also integrate `ASTR`, would this still be the case?

File: applications/Archisinal.md [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+| 5. | Wallet Functionality | We will integrate wallet functionality that is easy and intuitive, allowing users to connect their wallets and view their balances. |
+| 6. | Project Creation and NFT Minting | We will develop a user flow that allows architects to create a project and mint an NFT for their design. |
+| 7. | Buy Feature | We will implement a feature that allows users to purchase NFTs directly from the marketplace. |
+| 8. | Back-end Layer | We will develop simple node.js back-end layer for interaction with UI and orchestrating and indexing users and projects related data. |
+| 9. | UI/UX Design | We will implement UI/UX that reflects the marketplace's focus on architecture, engineering, and design, using visual elements and layouts tailored for the target audience. This will include a minimalist and mono-chrome theme, with a focus on showcasing the NFTs. |
+| 10. | Web app testing and deployment | We will implement E2E tests to cover basic user flows and deploy Web app on production. |
+| 11. | Production Network Deployment | Deploy the developed smart contracts on a production - Shiden network on Kusama. |
+
+...
+
```
Comment (HCastano) [inline applications/Archisinal.md:246]:
```suggestion
```

File: applications/Archisinal.md [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+- **Total Costs:**
+    - MVP: $60,000
+
+## MVP (brief discussion)
+
+### **Milestone 1: Smart Contract Development**
+
+- **Estimated duration:** 1 month
+- **FTE:** 3
+- **Costs:** 24,000 USD
```
Comment (HCastano) [inline applications/Archisinal.md:188]: This cost doesn't quite add up for me. 

You have two developers and one designer. Of these, only one has Rust experience. Let's assume that you're going to get the front-end developer writing Rust contracts, that's still only two FTEs that are needed for this smart-contract only milestone.

File: applications/Archisinal.md (hunk 10 above)
Comment (Archisinal) [reply applications/Archisinal.md:220]: Hi @HCastano - yes it would.

File: applications/Archisinal.md (hunk 12 above)
Comment (sventime) [reply applications/Archisinal.md:188]: Thank you, @HCastano, for your valid input. You're absolutely right that our UI/UX designer won't be involved in the 1st milestone. Let me provide more details about the responsibilities of our team members for the 1st milestone:

1. As a Rust engineer, my primary focus will be on the smart contracts part.
2. Our front-end engineer, Oleg, has prior experience in writing and testing smart contracts, and his main focus for this milestone will be on TypeScript integration tests.
3. Tom (@Archisinal) will be engaged in creating the architecture and will also be involved in the review process.

For the 2nd milestone, our development team will consist of me (for the back-end part), Oleg (for the front-end), and Artem (UI/UX design).

Comment (HCastano) [reply applications/Archisinal.md:188]: I hope the Rust code also has some integration tests and you don't rely on just the TS ones 😉 

Alright, thanks for clarifying 👍

**Review by HCastano — APPROVED** [review]
Alright I've gotten all my questions answered. Good luck!

**Comment by Archisinal** [comment]
> Alright I've gotten all my questions answered. Good luck!

Thanks for the support, Hernando!

**Review by 0xMarkian — APPROVED** [review]
Hey, I am approving this proposal. I believe it's an interesting idea for a real-world blockchain application. After our conversation with Tom, it's clear that the team has got relevant experience in the construction industry, is capable to technically delivery and is dedicated to making business results. 
That said, on milestone deliveries except for dev results, I would like to also see first-market results, customer development, a clear understanding of user audience(both seller and buyer on the marketplace), some preliminary user traction, and so on.

**Comment by 0xMarkian** [comment]
@Archisinal congratulations, your proposal has been officially accepted!

**Comment by TtomaS7** [comment]
Congratulations! We are thrilled to officially welcome you to the Ink!ubator program!! We can't wait to see the remarkable contributions you'll bring to the ecosystem.
In terms of the next steps, we are eagerly anticipating your first delivery, which is due in 1 month. When it’s ready, please create a PR here - https://github.com/use-inkubator/Grant-Milestone-Delivery

If you have any questions, concerns, or if you just want to brainstorm ideas, don't hesitate to reach out. We encourage you to utilize our Element channel for any inquiries - https://matrix.to/#/#sm-bounty-support:matrix.org

As you join the Ink!ubator family, please remember that you are not alone on this journey. We are here to support you every step of the way. We believe in your potential and we can't wait to see where this journey takes us all.

Congratulations once again and welcome aboard!
